### PR TITLE
New directory structure

### LINKED ISSUE #3683: Homestead accounts/contracts displayed while running Ropsten
parity started with `--chain ropsten`

I see the Ropsten accounts/contracts but also the Homestead ones.

screen shot 2016-11-29 at 20 02 37

### PR DESCRIPTION
**Pull request by arkpar**
This PR introduces a new directory structure for Paritys' files. The new structure looks like this:
```
.parity
  chains
    ethereum
      db
        906a34e69aec8c0d
          snapshot
          overlayrecent
            version
            db
              *.sst
          archive
            version
            db
              *.sst
      network
        nodes.json
      user_defauls	
    classic
    test
  keys
    ethereum
    test
    classic
  signer
  dapps
  ipc
  ```
* `--db-path` and `-d` now set a root path for all files (`.parity` alternative).
* `db kill` command is introduced.

Closes #3683, #1221, #2383, #2374

### CODE REVIEW

**Comment by rphmeier** [comment]
Seems to also close https://github.com/ethcore/parity/issues/2383

This seems much better. I assume the config file will go in `~/.parity/` root as well?

File: ethcore/src/spec/spec.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ pub struct Spec {
 	/// What engine are we using for this?
 	pub engine: Arc<Engine>,
 	/// The fork identifier for this chain. Only needed to distinguish two chains sharing the same genesis.
-	pub fork_name: Option<String>,
+	pub data_dir: String,
```
Comment (rphmeier) [inline ethcore/src/spec/spec.rs]: This seems to break compatibility with previous chain specs.
Also the doc string here is now wrong.

File: parity/upgrade.rs [hunk 2]
```diff
@@ -126,3 +129,76 @@ pub fn upgrade(db_path: Option<&str>) -> Result<usize, Error> {
 		upgrade_from_version(ver)
 	})
 }
+
+
```
Comment (rphmeier) [inline parity/upgrade.rs]: nit: whitespace

File: ethcore/src/spec/spec.rs (hunk 1 above)
Comment (arkpar) [inline ethcore/src/spec/spec.rs]: Don't think anyone uses this for private chains. We use it for classic and expanse. Upgrade paths for these is supported.

**Comment by arkpar** [comment]
Config goes to `~/.parity` by default.

**Comment by gavofyork** [comment]
would be good to solve https://github.com/ethcore/parity/issues/1648 in this PR too since they both need paths sorting.

**Comment by rphmeier** [comment]
having `--db-path` default set using `app_dirs` crate seems simple enough:
```rust
let app_info = AppInfo { name: "parity", author: "parity" };
let db_path = try!(get_app_root(AppDataType::UserData, &app_info));
```

**Comment by rphmeier** [comment]
It even obeys XDG base dir standards :)

**Comment by gavofyork** [comment]
maybe make `-d` -> `--root-path`; `--db-path` -> just for the chain/state databases; `--keys-path` -> for keys, ...

**Pull request merged**